File: cs/fib_loop.cs
using System;
using System.Threading;
using System.Globalization;
using Mono.Math;

public class Program
{
    static BigInteger Fibonacci(ulong n)
    {
        BigInteger a = 0;
        BigInteger b = 1;
        ulong i = 1;
        while( i <= n )
        {
            BigInteger c = a + b;
            a = b;
            b = c;
            i += 1;
        }
        return a;
    }

    public static void Main(string[] args)
    {
        Thread.CurrentThread.CurrentCulture = new CultureInfo("en-US");
        
        ulong iterations = args.Length > 0 ? Convert.ToUInt64(args[0]) : 100;
        ulong n = args.Length > 1 ? Convert.ToUInt64(args[1]) : 10;

        Timeit timeit = new Timeit();
        BigInteger result = timeit.Run<ulong, BigInteger>(iterations, Fibonacci, n);

        Reporter.write_result(Console.OpenStandardOutput(),
                                String.Format("Fibonacci (recursive) n = {0}", n),
                                iterations,
                                timeit.Average,
                                timeit.Mean,
                                timeit.Min,
                                timeit.Max,
                                String.Format("{0}", result) );
    }
}

File: python/fib_loop.py
import sys
from timeit import Timeit
import reporter

_range = xrange if sys.version_info.major == 2 else range


def Fibonacci(n):
    a = 0
    b = 1
    c = 0
    i = 1
    while i <= n:
        c = a + b
        a = b
        b = c
        i += 1
    return a

iterations = int(sys.argv[1]) if len(sys.argv) > 1 else 100
n = int(sys.argv[2]) if len(sys.argv) > 2 else 10

timeit = Timeit()

result = timeit.run(iterations, Fibonacci, n)

reporter.write_result(sys.stdout,
                      "Fibonacci (recursive) n = %d" % n,
                      iterations,
                      timeit.average,
                      timeit.mean,
                      timeit.min,
                      timeit.max,
                      str(result))
